.github/workflows/verify.yml

```diff
@@ -7,23 +7,22 @@ on:
     branches: [ main ]
 
 jobs:
-  build:
+  test-lint:
     runs-on: ubuntu-latest
-    strategy:
-      matrix:
-        python-version: ["3.8"] # The tests and lint is running inside venv with 2.7 anyway
+    container:
+      image: python:2.7
     steps:
     - uses: actions/checkout@v3
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v3
-      with:
-        python-version: ${{ matrix.python-version }}
+    - name: Install requirements
+      run: pip install --upgrade -r requirements.txt
     - name: Lint
-      run: make lint
+      run: pylint --rcfile=.pylintrc scripts/* tests/*
     - name: Test
-      run: make tests
-    - name: Upload coverage to Codecov
+      run: python -m pytest --cov --cov-report=xml
+    - name: Upload coverage reports to Codecov
       uses: codecov/codecov-action@v3
+      env:
+        CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
```

Comment:
For public repos we don't need codecov_token specified, I believe. Also, I may not be 100% right, but I guess that secret won't work if the PR is open from a fork, since secrets are not used in normal workflows... One would need to use a special pull_request workflow that has access to secrets. 

I would say that this is not needed for now.

```suggestion
```

Sometimes codecov will be mad, but it will upload the results nonetheless

Reply:
Alright, removed